DAQ_System/Source/main(updated).cpp

```diff
@@ -0,0 +1,120 @@
+                        if( i == n )
+                        {
+                            average[0] = sum[0]/n;
+                            average[1] = sum[1]/n;
+                            average[2] = sum[2]/n;
+                        }
+                    }
+
+
+                    offsetStatus = true; // offset is set so this setting makes it so it wont callibrate every single time.
+                }
+
+                // the equation used to find the dps of the gyroscope is dps = (rawdata - offset)*sensitivity
+                callibrated[0] = (gyroData[0] - average[0])*sensitivity;
+                callibrated[1] = (gyroData[1] - average[1])*sensitivity;
+                callibrated[2] = (gyroData[2] - average[2])*sensitivity;
+                // this is the conversion to radians per second
+                rads[0] = DPS_TO_RadS *callibrated[0];
+                rads[1] = DPS_TO_RadS *callibrated[1];
+                rads[2] = DPS_TO_RadS *callibrated[2];
+              
+                
+
+            
+
+                /* this is the ANSI escape sequences 
+             \033 is the escape character, which signals the start of an escape sequence. 
+            [2J is the clear screen command, which clears the entire screen.
+            [1;1H is the cursor position command, which moves the cursor to a specified 
+            
+               */
+                // raw data from the gyroscope
+                printf("\033[2J\033[1;1H"); 
+                printf("tx: %d \n",gyroData[0] ); 
+                printf("ty: %d \n",gyroData[1] );
+                printf("tz: %d \n",gyroData[2] );
+                // displays the average offset used for callibration of the gyroscope
+                printf("the average of the X-axis is: %d \n", average[0]);
+                printf("the average of the y-axis is: %d \n", average[1]);
+                printf("the average of the z-axis is: %d \n", average[2]);
+                // displays the callibrated data of the gyroscope in dps
+                printf("callibration of X(in dps): %d \n",callibrated[0]);
+                printf("callibration of Y(in dps): %d \n",callibrated[1]);
+                printf("callibration of Z(in dps): %d \n",callibrated[2]);
+                // displays the callibrated data of the gyroscope in radians per second
+                printf("callibration of x(in rad/s) %.5f \n",rads[0]);
+                printf("callibration of y(in rad/s) %.5f \n",rads[1]);
+                printf("callibration of z(in rad/s) %.5f \n",rads[2]);
+                // this is still under testing 
+                
+
+                thread_sleep_for(500);
+
+            
+
+                
+            }
+        
+            while (!Gyro.read(gyroData)) // when the gyroscope fails to read data then the this part of the code will execute until a successfull read is detected.
+            {
+                printf( "It's not Gyroin Time =( \r");
+
+            }
```

Comment:
The loop is unnecessary. Simply add the printf statement to outside the `while (Gyro.read(gyroData))` loop declared earlier